jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/test/BPMN2DataServicesReferencesTest.java

```diff
@@ -510,7 +521,41 @@ public void testSignalsAndGlobals() throws IOException {
                     "person".equals(globalName) || "name".equals(globalName) );
         }
 
+        assertNotNull("Signals description should not be null", procDef.getSignalsDesc());
+        assertNotNull("Messages description should not be null", procDef.getMessagesDesc());
+        assertTrue("Signal description should be empty", procDef.getSignalsDesc().isEmpty());
+        assertTrue("Messages description should be empty", procDef.getMessagesDesc().isEmpty());
+
         // cleanup
         processService.abortProcessInstance(procInstId);
     }
+
+    @Test
+    public void testSignalsDesc() throws IOException {
+        ProcessDefinition procDef = bpmn2Service.getProcessDefinition(deploymentId, "org.jbpm.boundarysignal");
+        assertNotNull(procDef);
+
+        Collection<SignalDesc> signals = procDef.getSignalsDesc();
+        assertNotNull("Signals description should not be null", signals);
+        assertFalse("Signal description should be empty", signals.isEmpty());
+
+        SignalDesc signal = signals.iterator().next();
+        assertEquals("_3b677877-9be0-3fe7-bfc4-94a862fdc919", signal.getId());
+        assertEquals("MySignal", signal.getName());
```

Comment:
I would add an assert for `structRef` as well, even if it is null, just to cover all possible attributes
```suggestion
        assertEquals("MySignal", signal.getName());
        assertNull("Structure Ref is expected to be null", signal.getStructureRef());
```